### ISSUE TITLE
Rayleigh sampling doesn't support broadcasting

### ISSUE THREAD

**Issue by NeilGirdhar** [google/jax#15320]
### Description

```python
import jax.numpy as jnp
from jax.random import rayleigh, PRNGKey

key = PRNGKey(123)
sigma = jnp.ones(10)
rayleigh(key, sigma, (3, 10))
```

I'm happy to submit a PR for this.

**Comment by NeilGirdhar** [comment]
@jakevdp Cool!

I actually test most of the sampling functions through my efax library.  I don't test dtypes, but I do test the broadcasting and the accuracy.  Should I submit a PR to fix this bug or just wait for your overhaul?
